### DISCUSSION TITLE
Label for instruments

### DISCUSSION THREAD

**Discussion by klkl0808** [microsoft/Qcodes#4411] [Ideas]
I'm writing this suggestion as a user of the quantify framework, which uses QCoDeS instruments in many places. (Even though this is related, this is a different suggestion than in [this discussion](https://github.com/QCoDeS/Qcodes/discussions/4289)).

When displaying instrument information to the user, quantify refers to the instrument by the instrument name. There were no restrictions on the naming, so that these instrument names had a reasonably nice format (including spaces, angle brackets, ...). Since instrument names need to be valid identifiers (introduced in [this PR](https://github.com/QCoDeS/Qcodes/pull/4178)), we cannot display nice instrument names anymore.

I realize that quantify has been abusing the instrument name in the past. But I'm wondering if there is a use case for instrument labels. In the same way in which Parameters have labels that can be displayed nicely, could instruments also carry a label? This way, instrument information can be displayed nicely.

**Comment by jenshnielsen** [comment]
@klkl0808 If the object is to have a nice, formatted instrument name where you are free to use any character etc then that sounds like a good solution and I would be happy to review such a pr. 

However, using it for the issue discussed in https://github.com/QCoDeS/Qcodes/discussions/4289  I think that this would be the wrong solution and something more robust should be developed.

**Reply by klkl0808** [reply]
Thanks for the reply, @jenshnielsen. No, the question of displaying instrument labels is separate from the name-parsing issue discussed elsewhere. I created a PR in #4460 and would appreciate feedback.
